### PR TITLE
added compile time option to perform range checking on ManagedArray

### PR DESCRIPTION
**Pull request by rrsettgast**

### CODE REVIEW

**Comment by rrsettgast** [comment]
Hey...what happened to the travis tests?

File: src/ManagedArray.inl [hunk 1]
```diff
@@ -207,6 +207,9 @@ template<typename T>
 template<typename Idx>
 CHAI_INLINE
 CHAI_HOST_DEVICE T& ManagedArray<T>::operator[](const Idx i) const {
+#if defined(CHAI_ARRAY_BOUNDS_CHECK)
+  assert( i>=0 && static_cast<size_t>(i) < m_elems );
```
Comment (davidbeckingsale) [inline src/ManagedArray.inl]: Can you make assert work on the GPU? I have a branch with this functionality, but couldn't get it working.

**Comment by davidbeckingsale** [comment]
https://github.com/LLNL/CHAI/tree/feature/CHAI-44-debug-range-check

File: src/ManagedArray.inl (hunk 1 above)
Comment (rrsettgast) [reply src/ManagedArray.inl]: Sorry. I didn't see that. Where is HOST_DEVICE_ASSERT defined? I can't find it.

I actually hadn't tried it on the GPU yet....and I will need to change the test to launch a kernel....but I think there should be support??
https://docs.nvidia.com/cuda/cuda-c-programming-guide/#assertion

**Comment by artv3** [comment]
@rrsettgast , @davidbeckingsale , I tried out the branch and assert worked for me (on the GPU). One suggestion is to add the <cassert> header in ManagedArray.inl . I would add it myself but I don't have permissions to push onto this branch.

**Comment by rrsettgast** [comment]
@artv3 do you mean including "assert.h" in ManagedArray.inl?

**Comment by artv3** [comment]
right @rrsettgast,  ``assert.h`` or <cassert> would also work.

**Comment by davidbeckingsale** [comment]
Are you sure `cassert` works?

**Comment by artv3** [comment]
@davidbeckingsale , it did for me. I also used GNU-4.9 as my host compiler.

**Pull request closed without merging**